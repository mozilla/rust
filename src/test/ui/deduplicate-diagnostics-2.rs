// revisions: duplicate deduplicate
//[deduplicate] compile-flags: -Z deduplicate-diagnostics=yes

fn main() {
    match 0.0 {
        1.0 => {} //~ ERROR floating-point types cannot be used in patterns
                  //~| WARNING this was previously accepted
                  //~| ERROR floating-point types cannot be used in patterns
                  //~| WARNING this was previously accepted
        2.0 => {} //~ ERROR floating-point types cannot be used in patterns
                  //~| WARNING this was previously accepted
                  //[duplicate]~| ERROR floating-point types cannot be used in patterns
                  //[duplicate]~| WARNING this was previously accepted
        _ => {}
    }
}
