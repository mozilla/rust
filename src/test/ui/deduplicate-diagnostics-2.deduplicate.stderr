error: floating-point types cannot be used in patterns
  --> $DIR/deduplicate-diagnostics-2.rs:6:9
   |
LL |         1.0 => {}
   |         ^^^
   |
   = note: `#[deny(illegal_floating_point_literal_pattern)]` on by default
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #41620 <https://github.com/rust-lang/rust/issues/41620>

error: floating-point types cannot be used in patterns
  --> $DIR/deduplicate-diagnostics-2.rs:10:9
   |
LL |         2.0 => {}
   |         ^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #41620 <https://github.com/rust-lang/rust/issues/41620>

error: floating-point types cannot be used in patterns
  --> $DIR/deduplicate-diagnostics-2.rs:6:9
   |
LL |         1.0 => {}
   |         ^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #41620 <https://github.com/rust-lang/rust/issues/41620>

error: aborting due to 3 previous errors

